src/ddmd/parse.d

```diff
@@ -5845,7 +5845,7 @@ final class Parser(AST) : Lexer
                 if (token.value == TOKfinally)
                 {
                     nextToken();
-                    finalbody = parseStatement(0);
+                    finalbody = parseStatement(PSscope);
```

Comment:
Thank you for tracking this down! It looks very similar to https://issues.dlang.org/show_bug.cgi?id=17690, can you do that one as well?